### PR TITLE
Support configuration cache

### PR DESCRIPTION
**Pull request by hfhbd**

### CODE REVIEW

File: src/main/kotlin/com/alecstrong/grammar/kit/composer/GrammarKitComposerPlugin.kt [hunk 1]
```diff
@@ -2,23 +2,26 @@ package com.alecstrong.grammar.kit.composer
...
+
+    val rootDir = project.layout.projectDirectory.dir("src${File.separatorChar}main${File.separatorChar}kotlin")
+    rootDir.forBnfFiles { bnfFile ->
+      val name = bnfFile.toRelativeString(rootDir.asFile).replace(File.separatorChar, '_').dropLast(4)
+      val outputDirectory = project.layout.buildDirectory.dir("grammars${File.separatorChar}$name")
 
       val compose = project.tasks.register("createComposable${name}Grammar", BnfExtenderTask::class.java) {
-        it.source(bnfFile)
-        it.outputDirectory = outputDirectory
-        it.include("**${File.separatorChar}*.bnf")
```
Comment (hfhbd) [inline src/main/kotlin/com/alecstrong/grammar/kit/composer/GrammarKitComposerPlugin.kt:21]: Do you know the reason for `include('**/*.bnf')` in the bnfExtenderTask?

**Comment by hfhbd** [comment]
Draft: I didn't test it yet.

**Comment by hfhbd** [comment]
```
Caused by: org.gradle.api.GradleException: Error: Unable to initialize main class org.intellij.grammar.Main
Caused by: java.lang.NoClassDefFoundError: com/intellij/openapi/util/KeyedExtensionCollector
	at org.jetbrains.grammarkit.tasks.GenerateParserTask.generateParser(GenerateParserTask.kt:85)
```
I didn't change the parser...

**Comment by hfhbd** [comment]
Contains tests from #73

File: src/main/kotlin/com/alecstrong/grammar/kit/composer/GrammarKitComposerPlugin.kt (hunk 1 above)
Comment (AlecStrong) [reply src/main/kotlin/com/alecstrong/grammar/kit/composer/GrammarKitComposerPlugin.kt:31]: incremental compilation, including the files as source is how it knows when to re-run the task

**Review by AlecStrong — APPROVED** [review]

File: src/main/kotlin/com/alecstrong/grammar/kit/composer/GrammarKitComposerPlugin.kt (hunk 1 above)
Comment (hfhbd) [reply src/main/kotlin/com/alecstrong/grammar/kit/composer/GrammarKitComposerPlugin.kt:31]: But we create 1 task for each bnf file by setting the `source` property. Why do we need to set include too? The input is only 1 bnf file. As long as we don't use multiple bnf files per task removing this code is fine.

**Comment by hfhbd** [comment]
BTW any update?

**Comment by AlecStrong** [comment]
forgot about this, I'll try and review it soon.

**Comment by hfhbd** [comment]
Any updates or free to merge?

**Review by AlecStrong — APPROVED** [review]

**Pull request merged**